### PR TITLE
chore: Add marketplace state

### PR DESCRIPTION
**Pull request by kaladivo**

### CODE REVIEW

**Comment by kaladivo** [comment]
closes #24

**Comment by kaladivo** [comment]
closes #25

File: apps/mobile/src/state/marketplace/index.ts [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+            },
+            (result) => result.offerIds
+          ),
+          (a) => a,
+          TE.fromTask
+        )
+      ),
+      TE.matchW(
+        (error) => {
+          store.set(loadingStateAtom, {state: 'error', error})
```
Comment (SamTremko) [inline apps/mobile/src/state/marketplace/index.ts:95]: report error

File: apps/mobile/src/state/marketplace/index.ts [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+export function useOffersLoadingError(): Option.Option<ApiErrorFetchingOffers> {
+  const offerState = useAtomValue(loadingStateAtom)
+
+  return offerState.state === 'error'
+    ? Option.some(offerState.error)
+    : Option.none
+}
+
+export function useGetOffers(): OneOfferInState[] {
```
Comment (SamTremko) [inline apps/mobile/src/state/marketplace/index.ts]: without get

File: apps/mobile/src/state/marketplace/index.ts [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+  return useAtomValue(offersToSee)
+}
+
+export function useGetMyOffers(): OneOfferInState[] {
```
Comment (SamTremko) [inline apps/mobile/src/state/marketplace/index.ts]: without get

File: apps/mobile/src/state/marketplace/index.ts [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+  return useAtomValue(myOffersAtom)
+}
+
+export function useGetSingleOffer(
```
Comment (SamTremko) [inline apps/mobile/src/state/marketplace/index.ts]: without get

File: apps/mobile/src/state/marketplace/index.ts [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+  offerId: OfferId
+): Option.Option<OneOfferInState> {
+  const foundOffer = useAtomValue(
+    useMemo(() => singleOfferAtom(offerId), [offerId])
+  )
+  return Option.fromNullable(foundOffer)
+}
+
+export function useGetOffersFiltered(filter: OffersFilter): OneOfferInState[] {
```
Comment (SamTremko) [inline apps/mobile/src/state/marketplace/index.ts]: without get

**Review by SamTremko — CHANGES_REQUESTED** [review]

**Review by SamTremko — APPROVED** [review]

**Comment by kaladivo** [comment]
close #25

**Pull request merged**